### DISCUSSION TITLE
GitHub repository_dispatch event trigger Provider

### DISCUSSION THREAD

**Discussion by modalkonform** [fluxcd/flux2#2122] [Proposals]
It would be great if you could provide an additional notification provider which would allow to trigger a GitHub workflow with a `repository_dispatch` event (`POST /repos/{owner}/{repo}/dispatches`).   

The use case for me would be to trigger a workflow of an external repository which would execute an end-to-end-test-suite after successful deployment of a service.

**Comment by alekspog** [comment]
@modalkonform I've created PR which should add new GitHub dispatch notification provider https://github.com/fluxcd/notification-controller/pull/369. You can check `docs/spec/v1beta1/provider.md` in the PR to see if it suits your needs. Please leave any comments if you have any other use cases for your GitHub workflows.
